### PR TITLE
feat: add events API endpoint interaction

### PR DESCRIPTION
**Pull request by sachin120**
Add Events API endpoint

### CODE REVIEW

File: entity/events.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+// Event represent an event for MAAS node
+type Event struct {
+	UserName    string `json:"username,omitempty"`
+	Node        string `json:"node,omitempty"`
+	Hostname    string `json:"hostname,omitempty"`
+	Created     string `json:"created,omitempty"`
+	Type        string `json:"type,omitempty"`
+	Description string `json:"description,omitempty"`
+	Level       string `json:"level,omitempty"`
```
Comment (skatsaounis) [inline entity/events.go:13]: As well:

```suggestion
	Level       event.LogLevel `json:"level,omitempty"`
```

Comment (skatsaounis) [reply entity/events.go:13]: plus the `go fmt` required after this suggestion

File: entity/event/types.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package event
+
+// LogLevel correlates to a log level for MAAS events.
+type LogLevel string
+
+// log levels refering from maas server https://github.com/maas/maas/blob/master/src/maasserver/models/eventtype.py#L25
```
Comment (skatsaounis) [inline entity/event/types.go:6]:
```suggestion
// Log levels referring from MAAS server https://github.com/maas/maas/blob/master/src/maasserver/models/eventtype.py#L25
```

File: entity/events.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package entity
+
+import "github.com/maas/gomaasclient/entity/event"
+
+// Event represent an event for MAAS node
```
Comment (skatsaounis) [inline entity/events.go:5]:
```suggestion
// Event represents an event for MAAS node
```

File: entity/events.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	Type        string `json:"type,omitempty"`
+	Description string `json:"description,omitempty"`
+	Level       string `json:"level,omitempty"`
+	ID          int    `json:"id,omitempty"`
+}
+
+// EventsResp represents the MAAS Events endpoint
+type EventsResp struct {
+	Events []Event `json:"events,omitempty"`
+	Count  int     `json:"count,omitempty"`
```
Comment (skatsaounis) [inline entity/events.go:20]: You are missing the pagination fields here:

```suggestion
	Count  int     `json:"count,omitempty"`
	NextURI string  `json:"next_uri,omitempty"`
	PrevURI string  `json:"prev_uri,omitempty"`
```

**Review by skatsaounis — CHANGES_REQUESTED** [review]
Hi @sachin120. After reviewing your PR I found some issues that I commented/suggested a modification inline. Please take a look :)

File: entity/events.go (hunk 1 above)
Comment (sachin120) [reply entity/events.go:13]: if we don't keep some type for Level, any string will be accepted and the user needs to pass the value explicitly

File: entity/events.go (hunk 4 above)
Comment (sachin120) [reply entity/events.go]: I will try not to miss such an issue. Thanks @skatsaounis  :-)

**Review by skatsaounis — APPROVED** [review]
LGTM :tada: Thank you for yet another one!

**Pull request merged**